ligo/Makefile

```diff
@@ -0,0 +1,32 @@
+# SPDX-FileCopyrightText: 2020 TQ Tezos
+# SPDX-License-Identifier: MIT
+
+# Ligo executable
+LIGO ?= ligo
+
+# Compile code
+BUILD = $(LIGO) compile-contract --syntax cameligo
+
+# Where to put build files
+OUT ?= out
+
+.PHONY: all trivialDAO build clean clean_all
+
+all: build
+
+# Compile LIGO contract into its michelson representation.
+build: src
```

Comment:
The way we do in `stablecoin` (where the target is an actual generated file, not a phony target) looks like a more proper usage of Makefiles to me, but I don't insist on doing that way.

Reply:
Do you know how do they use their way? Just `make stablecoin.tz` (as in CI), or their approach allows something smarter?

Reply:
I don't think there is anything smarter, but in `Makefile` targets are supposed to be actual files produced by the corresponding commands https://www.gnu.org/software/make/manual/make.html#Rule-Introduction

> A target is usually the name of a file that is generated by a program; examples of targets are executable or object files. A target can also be the name of an action to carry out, such as ‘clean’ (see Phony Targets).

AFAIU there is no big advantage in non-phony targets, it just seems to be more conventional. Well, this convention is from the 20-th century and maybe nobody cares about it nowadays.

Reply:
I updated the Makefile, now it does not rebuild the contract if it has already been built. Though the Makefile now looks a bit strange :roll_eyes: because unlike in stablecoin, I put the produced file in gitignored `out` folder, not directly at the `ligo` folder.